README.md

```diff
@@ -109,4 +123,18 @@ A working example can be found in `examples/k8s`. The files inside manifests dir
     $ kubectl exec apache-demo-8479b98dd4-82jnn -c apache -- cat /path/to/secret/config
    key1=valueFromSecret
    key2=2ndValueFromSecret
-    ```
\ No newline at end of file
+    ```
+   
+## Advanced features
+### getValueByOrderedKeys
+This extension allows you to use dynamic keys in your secrets, which could happen if you use an external Secret operator for example.
+
+When is this useful? E.g. if you use keys inside Secrets to represent some kind of hierarchical configuration:
+```yaml
+value: my-value
+value_aws: aws-specific-value
+value_azure: azure-specific-value
+```
+
+Use the following line to access it: `value={{ getValueByOrderedKeys (index .Secrets "secretValues") "value_aws" "value" }}`
```

Comment:
The docs below already say "for the first matching key", so I'd suggest renaming the function to `getValueByFirstMatchingKey`, that feels a bit more intuitive to me.